src/core/thread/network_data_local.cpp

```diff
@@ -54,9 +54,54 @@ Local::Local(Instance &aInstance)
 }
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
-otError Local::AddOnMeshPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength, int8_t aPrf, uint8_t aFlags, bool aStable)
+otError Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
 {
-    return AddPrefix(aPrefix, aPrefixLength, NetworkDataTlv::kTypeBorderRouter, aPrf, aFlags, aStable);
+    uint16_t flags = 0;
+
+    if (aConfig.mPreferred)
+    {
+        flags |= BorderRouterEntry::kPreferredFlag;
+    }
```

Comment:
nit: can we increase code coverage using code below?
```suggestion
    flags |= aConfig.mPreferred ? BorderRouterEntry::kPreferredFlag : 0;
```

Reply:
Fine with `if ...`